src/bootstrap/min_config.rs

```diff
@@ -0,0 +1,430 @@
+use core::fmt;
+use std::{
+    cmp,
+    collections::HashMap,
+    env, fs,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+use serde_derive::Deserialize;
+
+use crate::{
+    cache::{Interned, INTERNER},
+    config::{self, set},
+    flags::Flags,
+    t,
+    util::output,
+};
+
+/// The bare minimum config, suitable for `bootstrap-shim`, but sharing code with the main `bootstrap` binary.
+/// We only want to include indispensable parts of configuration to have highest possible BC for bootstrap-shim.
+/// Prefer adding new fields to `bootstrap::config::Config` if possible.
```

Comment:
"if possible" feels like it's not very clear to me. For example, I see comments below about verbose and dry_run *not* being used in the shim, yet they are here. So that seems odd - why are they here?

Reply:
I believe it's clear enough. We simply say "the reason we have this minimal version is to avoid breaking backward compatibility". (for now it's written as BC, I guess that's why you thought it's not clear).